From eec41250c1c0b12daa3ea247c5b7049c0605218c Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 25 Aug 2014 07:52:13 -0400
Subject: [tools] perf: Use the DWARF unwind info only if loaded

Message-id: <1408953210-26343-246-git-send-email-jolsa@redhat.com>
Patchwork-id: 88883
O-Subject: [PATCH RHEL7.1 BZ1133083 245/322] perf tools: Use the DWARF unwind info only if loaded
Bugzilla: 1133083
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1133083
https://bugzilla.redhat.com/show_bug.cgi?id=1133083

upstream
========
commit d11416e76b3e2f60ed6cfa7c532d3b6777f66527
Author: Jean Pihet <jean.pihet@linaro.org>
Date: Mon Dec 16 17:43:14 2013 +0100

description
===========
Use the info only if it has been found in the .debug_frame section of
the ELF binary.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/util/unwind.c b/tools/perf/util/unwind.c
index 0efd539..416f22b 100644
--- a/tools/perf/util/unwind.c
+++ b/tools/perf/util/unwind.c
@@ -340,10 +340,10 @@ find_proc_info(unw_addr_space_t as, unw_word_t ip, unw_proc_info_t *pi,
  /* Check the .debug_frame section for unwinding info */
  if (!read_unwind_spec_debug_frame(map->dso, ui->machine, &segbase)) {
   memset(&di, 0, sizeof(di));
-  dwarf_find_debug_frame(0, &di, ip, 0, map->dso->name,
-           map->start, map->end);
-  return dwarf_search_unwind_table(as, ip, &di, pi,
-       need_unwind_info, arg);
+  if (dwarf_find_debug_frame(0, &di, ip, 0, map->dso->name,
+        map->start, map->end))
+   return dwarf_search_unwind_table(as, ip, &di, pi,
+        need_unwind_info, arg);
  }
 #endif
 
-- 
1.7.1